lighthouse-core/audits/csp-xss.js

```diff
@@ -88,20 +84,17 @@ class CspXss extends Audit {
   }
 
   /**
+   * @param {import('../lib/csp-evaluator').Finding[][]} syntaxFindings
    * @param {string[]} rawCsps
    * @return {LH.Audit.Details.TableItem[]}
    */
-  static collectSyntaxResults(rawCsps) {
+  static constructSyntaxResults(syntaxFindings, rawCsps) {
     /** @type {LH.Audit.Details.TableItem[]} */
     const results = [];
 
-    const syntaxFindingsByCsp = evaluateRawCspForSyntax(rawCsps);
-    for (let i = 0; i < rawCsps.length; ++i) {
-      const items = syntaxFindingsByCsp[i].map(this.findingToTableItem);
-      if (!items.length) {
-        items.push({description: str_(UIStrings.noSyntaxErrors)});
-      }
-
+    for (let i = 0; i < syntaxFindings.length; ++i) {
+      const items = syntaxFindings[i].map(this.findingToTableItem);
+      if (!items.length) continue;
```

Comment:
this change will skip printing the full CSP if there's no syntax errors right?

thx for that. i was also thinking the same as brendan.

Reply:
Correct.